cmd/syncer/main.go

```diff
@@ -17,21 +17,46 @@ limitations under the License.
 package main
 
 import (
+	"context"
 	"flag"
 	"os"
 
+	"github.com/kubernetes-csi/csi-lib-utils/leaderelection"
 	"k8s.io/klog"
+	k8s "sigs.k8s.io/vsphere-csi-driver/pkg/kubernetes"
 
 	metadatasyncer "sigs.k8s.io/vsphere-csi-driver/pkg/syncer"
 )
 
+var (
+	enableLeaderElection = flag.Bool("leader-election", false, "Enable leader election.")
+)
+
 // main is ignored when this package is built as a go plug-in.
 func main() {
 	klog.InitFlags(nil)
 	flag.Parse()
 	metadataSyncer := metadatasyncer.NewInformer()
-	if err := metadataSyncer.Init(); err != nil {
-		klog.Errorf("Error initializing Metadata Syncer")
-		os.Exit(1)
+	run := func(ctx context.Context) {
+		if err := metadataSyncer.Init(); err != nil {
+			klog.Errorf("Error initializing Metadata Syncer")
+			os.Exit(1)
+		}
+	}
+
+	if !*enableLeaderElection {
+		run(context.TODO())
+	} else {
+		k8sClient, err := k8s.NewClient()
+		if err != nil {
+			klog.Errorf("Creating Kubernetes client failed. Err: %v", err)
+			os.Exit(1)
+		}
+		lockName := "vsphere-syncer"
+		le := leaderelection.NewLeaderElection(k8sClient, lockName, run)
+
+		if err := le.Run(); err != nil {
+			klog.Fatalf("Error initializing leader election: %v", err)
```

Comment:
suggestion: `Fatalf()` prints a stack trace, and `Exitf()` does not. `Exitf()` may be more appropriate here.